src/object-store/src/cache_policy.rs

```diff
@@ -0,0 +1,124 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::num::NonZeroUsize;
+use std::ops::DerefMut;
+use std::sync::{Arc, RwLock};
+
+use async_trait::async_trait;
+use futures::future::BoxFuture;
+use lru::LruCache;
+use opendal::layers::CachePolicy;
+use opendal::raw::output::Reader;
+use opendal::raw::{Accessor, RpDelete, RpRead};
+use opendal::{ErrorKind, OpDelete, OpRead, OpWrite, Result};
+
+struct LruValue;
+
+#[derive(Debug)]
+pub struct LruCachePolicy {
+    lru_cache: Arc<RwLock<LruCache<String, LruValue>>>,
```

Comment:
Looks like it always uses write lock,so we don't need the `RwLock`. I think we can use tokio's Mutex. https://docs.rs/tokio/latest/tokio/sync/struct.Mutex.html#method.lock